shell/platform/embedder/embedder.h

```diff
@@ -1587,6 +1605,10 @@ typedef struct {
   FlutterPoint offset;
   /// The size of the layer (in physical pixels).
   FlutterSize size;
+
+  /// Extra information for the backing store that the embedder may
+  /// use during presentation.
+  FlutterBackingStorePresentInfo* backing_store_present_info;
```

Comment:
Eventually, we'll also need to pass preexisting damage (from embedder client to embedder) and `buffer_damage` + `frame_damage` (from embedder to client) to support partial repaint. I'm not entirely sure where these belong. Technically we don't even need them per buffer, they are only once for all layers. But I haven't yet figured out partial repaint with multiple layers, the tricky part there is we may need to make sure that we preserve z-order.

Reply:
I'm not very familiar with this area so I'll let others provide guidance :)

> I'm not sure why this would break the ABI though?

Apologies, I misspoke. My concern is that `FlutterBackingStore`'s contains a [union of structs](https://github.com/flutter/engine/blob/08aaa88bf67fcd5a18d4309d89c1e940a544e49a/shell/platform/embedder/embedder.h#L1542-L1553). We would break the ABI if we add a member after this union and then add members to structs within that union.

Reply:
The client sets struct size though. So if we add a new union field and client sets `backing_store.size = size_of(FlutterBackingStoreWithoutNewMember)` we could detect it and convert the struct after the callback.

Reply:
I'm guessing though a complication here is that we pass the struct later in present callback, but not the original struct - a copy that embedder makes. And it gets hairy if the embedder wants to update fields after the union. Still doable, but we may internally need to keep union {  FlutterBackingStore, FlutterBackingStoreLegacy } around and access them based on client provided size.

Reply:
Example of adding fields to backing store the preserve abi compatibility: 
https://gist.github.com/knopp/9dbabd64a0501113ad73464463a51031. Instead of using the backing store directly the embedder would use a proxy class that is aware of different backing store versions.

Reply:
I added commit moving the new field into `FlutterBackingStore` in a way that preverves the ABI even if we need to add another field to the union.

Reply:
And now I have removed the commit. It was a bad idea. I don't think there is any reasonable way to extend `FlutterBackingStore` right now.

Reply:
Agreed. I think there's no way to safely modify that struct at this point and it should be locked down.